Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsComponentMode.cpp

```diff
@@ -121,6 +132,741 @@ namespace PhysX
+                actionIdentifier,
+                actionProperties,
+                []
+                {
+                    auto componentModeCollectionInterface = AZ::Interface<AzToolsFramework::ComponentModeCollectionInterface>::Get();
+                    AZ_Assert(componentModeCollectionInterface, "Could not retrieve component mode collection.");
+
+                    componentModeCollectionInterface->EnumerateActiveComponents(
+                        [](const AZ::EntityComponentIdPair&, const AZ::Uuid&)
+                        {
+                            auto componentModeCollectionInterface =
+                                AZ::Interface<AzToolsFramework::ComponentModeCollectionInterface>::Get();
+                            AZ_Assert(componentModeCollectionInterface, "Could not retrieve component mode collection.");
+
+                            componentModeCollectionInterface->EnumerateActiveComponents(
+                                [](const AZ::EntityComponentIdPair& entityComponentIdPair, const AZ::Uuid&)
+                                {
+                                    JointsComponentModeRequestBus::Event(
+                                        entityComponentIdPair, &JointsComponentModeRequests::SetCurrentSubMode,
+                                        JointsComponentModeCommon::SubComponentModes::ModeType::SnapPosition);
+                                }
+                            );
+                        }
+                    );
+                }
+            );
+
+            actionManagerInterface->InstallEnabledStateCallback(
+                actionIdentifier,
+                []()
+                {
+                    auto componentModeCollectionInterface = AZ::Interface<AzToolsFramework::ComponentModeCollectionInterface>::Get();
+                    AZ_Assert(componentModeCollectionInterface, "Could not retrieve component mode collection.");
+
+                    bool isComponentActive = false;
+                    bool isAvailable = true;
+
+                    componentModeCollectionInterface->EnumerateActiveComponents(
+                        [&isComponentActive, &isAvailable](const AZ::EntityComponentIdPair& entityComponentIdPair, const AZ::Uuid&)
+                        {
+                            isComponentActive = true;
+
+                            bool isCurrentSubModeAvailable = false;
+                            JointsComponentModeRequestBus::EventResult(
+                                isCurrentSubModeAvailable,
+                                entityComponentIdPair,
+                                &JointsComponentModeRequests::IsCurrentSubModeAvailable,
+                                JointsComponentModeCommon::SubComponentModes::ModeType::SnapPosition);
+
+                            if (!isCurrentSubModeAvailable)
+                            {
+                                isAvailable = false;
+                            }
```

Comment:
suggestion: Maybe as a second path it would be nice to refactor some of this code to remove the duplication if possible (ideally with some helper static or private member functions for example)

Reply:
Added a helper for this function too!